examples/modules/app-manager.api.ts

```diff
@@ -1,6 +1,8 @@
 import { AppManagerApi, Api, PluginApi } from '../../src'
+import { SecurityHandler } from './permission-handler'
 
 export class RemixAppManager extends AppManagerApi {
+  public permissionHandler = new SecurityHandler()

```

Comment:
naming it `securityHandler` would make more sense I suppose?

Reply:
The abstract class is `PermisionHandler` that's why the attribute is called `permissionHandler`. Here `SecurityHandler` extends `PermissionHandler`, so it is _a kind of_ `PermissionHandler`.

> also I think this SecurityHandler should be a param of appManager right?

When you say "param", you mean a parameter of the constructor ? Right now it's an abstract attribute that needs to be implemented. But I'm open to other options.

Reply:
> you mean a parameter of the constructor 
 
yes

> it's an abstract attribute that needs to be implemented

so there's a `set` function for this?

Reply:
Hmmm, not sure to understand. I'll clean the code a little, and add more examples. Hopefully it's going to be easier to understand.

Reply:
i wasn't sure how you plan to be able to set a `PermissionHandler` (by having a `setPermissionHandler` or directly in the constructor) . But I think your last commits make it clear

Reply:
I think every method is ok, either in the constructor, or as a parameter, or set later with `app.permissionHandler = new PermissionHandler()`. It shoud work for all.